subsystem/src/main/java/org/wildfly/subsystem/resource/operation/AddResourceOperationStepHandler.java

```diff
@@ -0,0 +1,292 @@
+                        // Otherwise, we need a separate write-attribute operation
+                        ModelNode writeAttributeOperation = Util.getWriteAttributeOperation(targetAddress, targetAttribute.getName(), targetValue);
+                        ImmutableManagementResourceRegistration targetRegistration = (currentAddress == targetAddress) ? registration : context.getRootResourceRegistration().getSubModel(targetAddress);
+                        if (targetRegistration == null) {
+                            throw new OperationFailedException(ControllerLogger.MGMT_OP_LOGGER.noSuchResourceType(targetAddress));
+                        }
+                        AttributeAccess targetAttributeAccess = targetRegistration.getAttributeAccess(PathAddress.EMPTY_ADDRESS, targetName);
+                        if (targetAttributeAccess == null) {
+                            throw new OperationFailedException(ControllerLogger.MGMT_OP_LOGGER.unknownAttribute(targetName));
+                        }
+                        OperationStepHandler writeAttributeHandler = targetAttributeAccess.getWriteHandler();
+                        if (writeAttributeHandler == null) {
+                            throw new OperationFailedException(ControllerLogger.MGMT_OP_LOGGER.attributeNotWritable(targetName));
+                        }
+                        context.addStep(writeAttributeOperation, writeAttributeHandler, OperationContext.Stage.MODEL);
+                    }
+                }
+            }
+        }
+        // Validate and apply proper attributes to model
+        ModelNode model = resource.getModel();
+        Map<String, AttributeAccess> attributes = registration.getAttributes(PathAddress.EMPTY_ADDRESS);
+        for (AttributeDefinition parameter : operationDefinition.getParameters()) {
+            if (this.descriptor.getAttributeTranslation(parameter) == null) {
+                AttributeAccess attribute = attributes.get(parameter.getName());
+                if ((attribute != null) && AttributeAccess.Storage.CONFIGURATION.test(attribute)) {
+                    OperationStepHandler writeHandler = attribute.getWriteHandler();
+                    if ((writeHandler != null) && (writeHandler.getClass() != WriteAttributeOperationStepHandler.class)) {
+                        // If attribute has custom handling, perform a separate write-attribute operation
+                        ModelNode writeAttributeOperation = Util.getWriteAttributeOperation(currentAddress, parameter.getName(), parameter.validateOperation(operation));
+                        context.addStep(writeAttributeOperation, writeHandler, OperationContext.Stage.MODEL);
+                    } else {
+                        // Auto-populate add resource operation parameters that correspond to resource attributes
+                        parameter.validateAndSet(operation, model);
+                    }
+                    parameter.validateAndSet(operation, model);
```

Comment:
This doesn't seem right.

Reply:
Hmm.  No, it isn't.  Not sure how that snuck in there.  Will fix.

Reply:
Fixed.